node/src/Pos/Block/Network/Logic.hs

```diff
@@ -348,6 +344,41 @@ handleRequestedHeaders cont inRecovery headers = do
             logDebug msg
             throwM $ DialogUnexpected msg
   where
+    newestHeader = headers ^. _NewestFirst . _neHead
+    oldestHeader = headers ^. _NewestFirst . _neLast
+    newestHash = headerHash newestHeader
+    oldestHash = headerHash oldestHeader
+    oldestEpoch = oldestHeader ^. epochIndexL
+
+    tryCalculateLrc = do
+        tip <- DB.getTipHeader @ssc
+        let tipEpochOrSlot = tip ^. epochOrSlotG
+        let tipEpoch = tip ^. epochIndexL
+        let differentEpochs = oldestEpoch == tipEpoch + 1
+        -- CSL-1660 We should also check that there's k blocks after
+        -- crucial slot.
+        let tipAfterCrucial = case unEpochOrSlot tipEpochOrSlot of
+                -- we assume differentEpochs is true, so this is
+                -- before zero's slot of epoch e, while new header has
+                -- epoch e+1.
+                Left _   -> False
+                Right si -> siSlot si > siSlot (crucialSlot $ siEpoch si)
```

Comment:
It seems more natural to pass `siEpoch si + 1`, but it doesn't matter here, because you compare `siSlot`s eventually.